playbooks/operationalizing/ingress.adoc

```diff
@@ -0,0 +1,202 @@
+
+[source]
+----
+networkConfig:
+  ingressIPNetworkCIDR: 10.9.54.0/25
+----
+
+Restart the OpenShift master service to apply the changes.
+
+=== Deploy a sample application
+
+To expose a PostgreSQL as a service for external consumption, the application must be first deployed into the cluster. Create a new project or use and existing project and instantiate one of the PostgreSQL templates as shown below using the OpenShift Command Line tool.  
+
+[source]
+----
+oc new-app --template=postgresql-ephemeral
+----
+
+CAUTION: The _postgresql-ephemeral_ template does not make use of persistent storage. Once the application is scaled down or destroyed, any existing data will be lost. To use persistent storage, specify the `postgresql-persistent` template instead.
+
+After instantiating the template, a _ClusterIP_ based service and DeploymentConfig is created and a new pod containing PostgreSQL will be started.
+
+
+=== Configuring an IP Address for a Service
+
+To allow the cluster to automatically assign an IP address for a service, create a service definition similar to the following that will create a new ingress service:
+
+[source]
+----
+oc create -f - <<INGRESS
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgresql-ingress
+spec:
+  ports:
+  - name: postgresql
+    port: 5432
+  type: LoadBalancer <1>
+  selector:
+    name: postgresql
+INGRESS
+----
+<1> The LoadBalancer type of service will make the request for an external service on behalf of the application user 
+
+Alternatively, the `oc expose` command can be used to create the servie.
+
+[source]
+----
+oc expose dc postgresql --type=LoadBalancer --name=postgresql-ingress
+----
+ 
+Once the service has been created, note the external ip address has been automatically allocated by the cluster and can be confirmed by running `oc get svc postgresql-ingress`.
+
+[source]
+----
+NAME         CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
+postgresql-ingress    172.30.74.106   10.9.54.100,10.9.54.100    5432/TCP    30s
+----
+
+Specifying the type _LoadBalancer_ has also configured the service with a _nodePort_ value allowing direct access to the service on each of the node's the application is deployed on. To discover the node port automatically assigned, execute `oc export svc postgresql-ingress`
+
+[source]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: null
+  labels:
+    app: postgresql-persistent
+    template: postgresql-persistent-template
+  name: postgresql-ingress
+spec:
+  deprecatedPublicIPs:
+  - 10.9.54.100
+  externalIPs:
+  - 10.9.54.100
+  ports:
+  - nodePort: 32439 <1>
+    port: 5432
+    protocol: TCP
+    targetPort: 5432
+  selector:
+    name: postgresql
+  sessionAffinity: None
+  type: LoadBalancer
+----
+<1> Automatically assigned port
+
+A PostgreSQL client can now be configured to connect directly to the node the application is deploy on using the value of the assigned nodePort.
```

Comment:
Isn't it that every node exposes this port ? Or it this only true when the type is also `NodePort` ? (If so, maybe one could add a sentence to explain the different meaning of `nodePort` for type `NodePort` and `LoadBalancer`.